tests/typ/compiler/string.typ

```diff
@@ -126,6 +128,45 @@
 #test("123".replace(regex("\d$"), "_"), "12_")
 #test("123".replace(regex("\d{1,2}$"), "__"), "1__")
 
+// with `Func` replacements
+
+//  with regex
```

Comment:
the tests are a bit unorganized and wildly formatted, could you tidy them up in line with how the other tests are?